extensions/src/error/PersistenceError.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * Error thrown when trying to write MSAL cache to persistence.
+ */
+export class PersistenceError extends Error {
```

Comment:
Any reason you don't extend AuthError.ts here? That way you don't have to do all the constructor code below again.

Reply:
Interesting, had not considered - these errors are not auth errors though, they're instead solely related to persistence. I think it could be valuable to make that distinction.

Reply:
It's a good point, hadn't considered it. I think since these are not really auth errors (they are all related ), it's good to make the differentiation.

Reply:
I think different `error` class makes sense here.

Reply:
I guess I agree that a different error class makes sense here, but AuthError is meant to be a base class for all MSAL errors, not just auth based ones (I agree the name is misleading). Any way we can merge it so as not to have duplicated code? Things like setPrototypeOf should preferably not be duplicated across multiple files.

Reply:
Will make this change, as well as other improvements to errors, in follow up PR.